### DISCUSSION TITLE
want to install openssl

### DISCUSSION THREAD

**Discussion by pradiptasahoo** [openssl/openssl#24840] [Q&A] [answered]
I got the error:
/usr/local/bin/perl util/mkinstallvars.pl PREFIX=. BINDIR=apps LIBDIR= INCLUDEDIR=include APPLINKDIR=ms ENGINESDIR=engines MODULESDIR=providers "VERSION=3.4.0-dev" "LDLIBS=-ldl -pthread " > builddata.pm
"pairs" is not exported by the List::Util module
Can't continue after import errors at util/mkinstallvars.pl line 14.
BEGIN failed--compilation aborted at util/mkinstallvars.pl line 14.
make[1]: *** [builddata.pm] Error 255

output uof command:perl configdata.pm --dump

root@83b963a9a2c6:~/sdk/apps/private/openssl# perl configdata.pm --dump

Command line (with current working directory = .):

    /usr/local/bin/perl ./Configure

Perl information:

    /usr/local/bin/perl
    5.18.2 for x86_64-linux-gnu-thread-multi

Enabled features:

    afalgeng
    apps
    argon2
    aria
    asm
    async
    atexit
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    default-thread-pool
    deprecated
    des
    dgram
    dh
    docs
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    ecx
    engine
    err
    filenames
    gost
    http
    idea
    legacy
    loadereng
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    quic
    unstable-qlog
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm2-precomp
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    thread-pool
    threads
    tls
    integrity-only-ciphers
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests             [cascade]        OPENSSL_NO_ACVP_TESTS
    asan                   [default]        OPENSSL_NO_ASAN
    brotli                 [default]        OPENSSL_NO_BROTLI
    brotli-dynamic         [default]        OPENSSL_NO_BROTLI_DYNAMIC
    buildtest-c++          [default]        
    winstore               [not-windows]    OPENSSL_NO_WINSTORE
    crypto-mdebug          [default]        OPENSSL_NO_CRYPTO_MDEBUG
    demos                  [default]        OPENSSL_NO_DEMOS
    h3demo                 [default]        OPENSSL_NO_H3DEMO
    devcryptoeng           [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128    [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                    [default]        OPENSSL_NO_EGD
    external-tests         [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                   [default]        
    fips-securitychecks    [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl               [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer         [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                   [default]        OPENSSL_NO_KTLS
    makedepend             [unavailable]    
    md2                    [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                   [default]        OPENSSL_NO_MSAN
    pie                    [default]        OPENSSL_NO_PIE
    rc5                    [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                   [default]        OPENSSL_NO_SCTP
    tfo                    [default]        OPENSSL_NO_TFO
    trace                  [default]        OPENSSL_NO_TRACE
    ubsan                  [default]        OPENSSL_NO_UBSAN
    unit-test              [default]        OPENSSL_NO_UNIT_TEST
    uplink                 [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers       [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                   [default]        OPENSSL_NO_ZLIB
    zlib-dynamic           [default]        OPENSSL_NO_ZLIB_DYNAMIC
    zstd                   [default]        OPENSSL_NO_ZSTD
    zstd-dynamic           [default]        OPENSSL_NO_ZSTD_DYNAMIC
    ssl3                   [default]        OPENSSL_NO_SSL3
    ssl3-method            [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86_64",
    bin_cflags => "",
    bin_lflags => "",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread -m64",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR = mipsel-buildroot-linux-uclibc-ar
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = mipsel-buildroot-linux-uclibc-gcc -mips32r2 -msoft-float
    CFLAGS = 
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = 
    CPPINCLUDES = 
    CROSS_COMPILE = mipsel-buildroot-linux-uclibc-
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LD = mipsel-buildroot-linux-uclibc-ld
    LDFLAGS = 
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = mipsel-buildroot-linux-uclibc-ranlib
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = mipsel-buildroot-linux-uclibc-mipsel-buildroot-linux-uclibc-ar
    ARFLAGS         = qc
    ASFLAGS         = 
    CC              = mipsel-buildroot-linux-uclibc-mipsel-buildroot-linux-uclibc-gcc -mips32r2 -msoft-float
    CFLAGS          = -Wall -O3
    CPPDEFINES      = 
    CPPFLAGS        = 
    CPPINCLUDES     = 
    CROSS_COMPILE   = mipsel-buildroot-linux-uclibc-
    CXX             = mipsel-buildroot-linux-uclibc-g++
    CXXFLAGS        = -Wall -O3
    HASHBANGPERL    = /usr/bin/env perl
    LD              = mipsel-buildroot-linux-uclibc-mipsel-buildroot-linux-uclibc-ld
    LDFLAGS         = 
    LDLIBS          = 
    PERL            = /usr/local/bin/perl
    RANLIB          = mipsel-buildroot-linux-uclibc-mipsel-buildroot-linux-uclibc-ranlib
    RC              = mipsel-buildroot-linux-uclibc-windres
    RCFLAGS         = 

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
root@83b963a9a2c6:~/sdk/apps/private/openssl#

**Comment by nhorman** [comment]
@levitte thoughts here?  Seems like a broken install of List::Util I think

**Comment by jon-oracle** [comment]
I'm getting this same error too on one of our Solaris Sparc hosts. Could be related to perl version? That particular machine is running perl 5.12.5.

What is the recommended minimum perl version for OpenSSL these days?

**Reply by tkoecker** [reply]
I've encountered the same issue on our Solaris boxes.
According to https://metacpan.org/pod/List::Util "pairs" was introduced in "List::Util" version 1.29.
According to https://perldoc.perl.org/5.20.0/List::Util the oldest perl release that contains this change (it contains "List::Util" version 1.38) is perl 5.20.0.
The previous release (perl 5.18.4) still contained "List::Util" version 1.27.

**Comment by levitte** [comment, marked as answer]
There's a fix: #25367

**Reply by levitte** [reply]
It's unfortunate that this didn't reach my attention earlier 😖
